conn.go

```diff
@@ -36,11 +36,27 @@ func (pr *PacketResponse) ReadPacket() (*ber.Packet, error) {
 	return pr.Packet, pr.Error
 }
 
+type messageContext struct {
+	id        int64
+	done      chan struct{}
+	responses chan *PacketResponse
+}
+
+func (msgCtx *messageContext) sendResponse(packet *PacketResponse) {
```

Comment:
doc that this should only be done within the processMessages loop... we can't depend on the finishMessage-triggered `close(done)` taking effect before the processMessages-triggered `close(responses)` if this is called from a third goroutine, which would mean we'd have a ~50% chance of a panic trying to send a packet to a closed channel